From dddea366c63efca18ae01dc88c2cf6333a630198 Mon Sep 17 00:00:00 2001
From: ejaaskel <esa.jaaskela@suomi24.fi>
Date: Fri, 14 Nov 2025 08:49:22 +0200
Subject: [PATCH] policy/modules/system/ssh: Allow changing password

Allow changing an expired password with an SSH connection. This requires
accesses to the etc and passwd for the ssh_t type. Note that allowing
remote password changes can be a security risk, and it should be
considered carefully.

Upstream-Status: Inappropriate [Sulka specific]
---
 policy/modules/services/ssh.te | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/policy/modules/services/ssh.te b/policy/modules/services/ssh.te
index 2d10ecb73..8c9e61901 100644
--- a/policy/modules/services/ssh.te
+++ b/policy/modules/services/ssh.te
@@ -1,5 +1,9 @@
 policy_module(ssh)
 
+gen_require(`
+	type passwd_exec_t;
+')
+
 ########################################
 #
 # Declarations
@@ -106,6 +110,12 @@ allow ssh_t sshd_tmp_t:dir manage_dir_perms;
 allow ssh_t sshd_tmp_t:file manage_file_perms;
 files_tmp_filetrans(ssh_t, sshd_tmp_t, { file dir })
 
+# Expired password change with SSH
+allow sshd_t passwd_exec_t:file { map execute execute_no_trans open read };
+allow sshd_t self:process setfscreate;
+files_manage_etc_files(sshd_t)
+auth_manage_shadow(sshd_t)
+
 manage_files_pattern(ssh_t, ssh_tmpfs_t, ssh_tmpfs_t)
 manage_lnk_files_pattern(ssh_t, ssh_tmpfs_t, ssh_tmpfs_t)
 manage_fifo_files_pattern(ssh_t, ssh_tmpfs_t, ssh_tmpfs_t)
-- 
2.34.1

